spec/views/user_show_spec.rb

```diff
@@ -0,0 +1,55 @@
+require 'rails_helper'
+
+RSpec.describe 'users#show', type: :feature do
```

Comment:
- [ ] Hey there! 🌟 When running tests for the show page, I encountered some errors. The main culprits seem to be variable naming inconsistencies. For instance, in your Users class, there's a method named most_recet_posts that fetches the three most recent posts. However, in the test, you're calling it as recent_posts. 😬

Another issue I noticed is related to private helper methods. While you've created these helpers, there's no callback method allowing post action controllers to access them. As a result, a nil class error pops up. 🚨

Consider updating the method names for consistency and ensure that the necessary callbacks are in place to resolve these hiccups. Happy coding! 🚀

![image](https://github.com/BilalLiaquat7/blogapp/assets/112261237/7d465e2c-0d97-4085-8c5a-bf2bd536dead)